Author: Hugh Greenberg <hugegreenbug@gmail.com>
Desc: Fix for the mouse being grabbed when clicking directly underneath
the title bar. Also, changes to disable the compositor.

diff -urN xfwm4-4.11.1/debian/rules xfwm4-4.11.1.new/debian/rules
--- xfwm4-4.11.1/debian/rules	2014-04-09 03:14:41.000000000 -0400
+++ xfwm4-4.11.1.new/debian/rules	2015-01-12 18:52:47.690511033 -0500
@@ -8,7 +8,7 @@
 	dh_strip --dbg-package=xfwm4-dbg
 
 override_dh_auto_configure:
-	dh_auto_configure -- --disable-silent-rules
+	dh_auto_configure -- --disable-silent-rules --disable-compositor
 
 %:
 	dh $@ --parallel
diff -urN xfwm4-4.11.1/disable_compositor xfwm4-4.11.1.new/disable_compositor
--- xfwm4-4.11.1/disable_compositor	1969-12-31 19:00:00.000000000 -0500
+++ xfwm4-4.11.1.new/disable_compositor	2015-01-12 18:52:44.414511091 -0500
@@ -0,0 +1,27 @@
+diff -urN xfwm4-4.11.1/debian/rules xfwm4-4.11.1.new/debian/rules
+--- xfwm4-4.11.1/debian/rules	2014-04-09 03:14:41.000000000 -0400
++++ xfwm4-4.11.1.new/debian/rules	2014-11-27 14:20:53.728574035 -0500
+@@ -8,7 +8,7 @@
+ 	dh_strip --dbg-package=xfwm4-dbg
+ 
+ override_dh_auto_configure:
+-	dh_auto_configure -- --disable-silent-rules
++	dh_auto_configure -- --disable-silent-rules --disable-compositor
+ 
+ %:
+ 	dh $@ --parallel
+diff -urN xfwm4-4.11.1/settings-dialogs/tweaks-settings.c xfwm4-4.11.1.new/settings-dialogs/tweaks-settings.c
+--- xfwm4-4.11.1/settings-dialogs/tweaks-settings.c	2013-12-26 16:41:22.000000000 -0500
++++ xfwm4-4.11.1.new/settings-dialogs/tweaks-settings.c	2014-11-27 14:21:38.724573230 -0500
+@@ -444,6 +444,11 @@
+     vbox = GTK_WIDGET (gtk_builder_get_object (builder, "main-vbox"));
+     gtk_widget_show_all (vbox);
+ 
++    /* Hack to hide the compositor tab */
++    GtkWidget *ctab;
++    ctab = GTK_WIDGET (gtk_builder_get_object (builder, "vbox7"));
++    gtk_widget_hide(ctab);
++
+     g_free (easy_click);
+     g_free (activate_action);
+ }
diff -urN xfwm4-4.11.1/settings-dialogs/tweaks-settings.c xfwm4-4.11.1.new/settings-dialogs/tweaks-settings.c
--- xfwm4-4.11.1/settings-dialogs/tweaks-settings.c	2013-12-26 16:41:22.000000000 -0500
+++ xfwm4-4.11.1.new/settings-dialogs/tweaks-settings.c	2015-01-12 18:52:47.690511033 -0500
@@ -444,6 +444,11 @@
     vbox = GTK_WIDGET (gtk_builder_get_object (builder, "main-vbox"));
     gtk_widget_show_all (vbox);
 
+    /* Hack to hide the compositor tab */
+    GtkWidget *ctab;
+    ctab = GTK_WIDGET (gtk_builder_get_object (builder, "vbox7"));
+    gtk_widget_hide(ctab);
+
     g_free (easy_click);
     g_free (activate_action);
 }
diff -urN xfwm4-4.11.1/src/netwm.c xfwm4-4.11.1.new/src/netwm.c
--- xfwm4-4.11.1/src/netwm.c	2013-12-26 16:41:22.000000000 -0500
+++ xfwm4-4.11.1.new/src/netwm.c	2015-01-12 18:53:47.130509968 -0500
@@ -695,9 +695,7 @@
             resize = TRUE; /* Resize */
             break;
         case NET_WM_MOVERESIZE_MOVE:
-            event->type = ButtonPress;
             resize = FALSE; /* Move */
-            break;
         case NET_WM_MOVERESIZE_CANCEL:
             FLAG_UNSET (c->xfwm_flags, XFWM_FLAG_MOVING_RESIZING);
             /* Walk through */
